spec/lib/efile/nj/nj_retirement_income_helper_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+RSpec.describe Efile::Nj::NjRetirementIncomeHelper do
+  describe ".calculate_maximum_exclusion" do
+    [
+      { total_income: 99_000, expected: 75_000 },
+      { traits: [:head_of_household], total_income: 99_000, expected: 75_000 },
+      { traits: [:qualifying_widow], total_income: 99_000, expected: 75_000 },
+      { traits: [:married_filing_jointly], total_income: 99_000, expected: 100_000 },
+      { traits: [:married_filing_separately], total_income: 99_000, expected: 50_000 },
+
+      { total_income: 120_000, expected: 45_000 },
+      { traits: [:head_of_household], total_income: 120_000, expected: 45_000 },
+      { traits: [:qualifying_widow], total_income: 120_000, expected: 45_000 },
+      { traits: [:married_filing_jointly], total_income: 120_000, expected: 60_000 },
+      { traits: [:married_filing_separately], total_income: 120_000, expected: 30_000 },
+
+      { total_income: 130_000, expected: 24_375 },
+      { traits: [:head_of_household], total_income: 130_000, expected: 24_375 },
+      { traits: [:qualifying_widow], total_income: 130_000, expected: 24_375 },
+      { traits: [:married_filing_jointly], total_income: 130_000, expected: 32_500 },
+      { traits: [:married_filing_separately], total_income: 130_000, expected: 16_250 },
+
+      { total_income: 150_001, expected: 0 },
+      { traits: [:head_of_household], total_income: 150_001, expected: 0 },
+      { traits: [:qualifying_widow], total_income: 150_001, expected: 0 },
+      { traits: [:married_filing_jointly], total_income: 150_001, expected: 0 },
+      { traits: [:married_filing_separately], total_income: 150_001, expected: 0 },
+    ].each do |test_case|
+      context "when filing with #{test_case}" do
+        let(:intake) do
+          create(:state_file_nj_intake, *test_case[:traits])
+        end
+        it "returns #{test_case[:expected]}" do
+          helper = Efile::Nj::NjRetirementIncomeHelper.new(intake)
+          result = helper.calculate_maximum_exclusion(test_case[:total_income])
+          expect(result).to eq(test_case[:expected])
+        end
+      end
+    end
+
+    describe ".eligible?" do
+      [
+        { traits: [:mfj_spouse_over_65], total_income: 99_000, earned_income: 3000, line_28a: 0, expected: true },
+        { traits: [:primary_over_65], total_income: 99_000, earned_income: 3000, line_28a: 0, expected: true },
+        { total_income: 99_000, earned_income: 3000, line_28a: 0, expected: false },
+        { traits: [:primary_over_65], total_income: 160_000, earned_income: 3000, line_28a: 0, expected: false },
+        { traits: [:primary_over_65], total_income: 99_000, earned_income: 4000, line_28a: 0, expected: false },
+        { traits: [:primary_over_65], total_income: 99_000, earned_income: 4000, line_28a: 100_000, expected: false },
```

Comment:
[pebble] Similar note about testing edge cases. Regarding age, I feel weird about using the over-65 trait when our logic here is checking the over-62 case. Could we add a new trait for over/under 62 so that we can verify that logic? If we changed the magic number in our code from 62 to 63, this test would still pass

Reply:
And similarly, we could gain more confidence by using 3,001 instead of 4,000 for the line-15 criteria, and $99,999, $150,000, and $150,001 for the total income criteria. This would provide a lot more confidence in the true logical behavior of the code, and also be a little more descriptive of the logic just from reading the cases